### ISSUE TITLE
Driver: System.Console driver

### ISSUE THREAD

**Issue by migueldeicaza** [migueldeicaza/gui.cs#27]
Currently we only have the ncurses driver, we need a Console driver.

**Comment by fanoI** [comment]
I think it will be great to have a fully cross platform without dependency to ncurses so it could work on OS that have neither ncurses neither a C API (as Cosmos).

**Comment by migueldeicaza** [comment]
Right, we need to auto-detect the platform at startup and choose the right driver, currently it is hardcoded to NCurses and I change it when I work there.

https://github.com/migueldeicaza/gui.cs/blob/c5325c55272d5e35b824e99076ad1c67d7719be3/Terminal.Gui/Core.cs#L1160

**Comment by fanoI** [comment]
In you opinion C# Console can do all NCurses does?

**Comment by fanoI** [comment]
Thank you! I'll test this soon...
